### PR TITLE
Message collections

### PR DESCRIPTION
**Pull request by rgbkrk**
This aggregates collections of messages after a kernel is launched.

screen shot 2018-12-21 at 10 28 16 pm

Since I want to help get `master` unstuck for app builds, I'm just posting this for others to look at.

### CODE REVIEW

File: packages/kernel-relay/src/index.ts [hunk 1]
```diff
@@ -12,12 +18,18 @@ const Types = gql`
...
+    id: ID!
+    payload: JSON
+  }
 `;
 
 const Query = gql`
   type Query {
     listKernelSpecs: [KernelSpec!]!
     running: [KernelSession!]!
+    messages: [Message!]!
```
Comment (captainsafia) [inline packages/kernel-relay/src/index.ts:32]: What do the two exclamation points mean here? You can have null value in the array?

File: packages/kernel-relay/src/index.ts [hunk 2]
```diff
@@ -33,27 +45,65 @@ type StartKernel = {
...
+      const subscription = kernel.channels.subscribe(
+        (message: JupyterMessage) => {
+          messages[id].push(message);
+          console.log(message);
+        }
+      );
+
+      // THOUGHT: Should we track our own messages sent out too?
+      //          Hmmmm
+      kernel.channels.next(kernelInfoRequest());
```
Comment (captainsafia) [inline packages/kernel-relay/src/index.ts]: Me thinks we should track outbound messages.

File: packages/kernel-relay/src/index.ts (hunk 1 above)
Comment (rgbkrk) [reply packages/kernel-relay/src/index.ts:32]: It means you can't have null values in the array and the array itself can't be null.

File: packages/kernel-relay/src/index.ts (hunk 2 above)
Comment (rgbkrk) [reply packages/kernel-relay/src/index.ts]: I agree. Oooh. This makes me realize that some of the magic that enchannel does we'll need to explicitly set here. Enchannel applies generated session IDs and other header info if not set.

Comment (rgbkrk) [reply packages/kernel-relay/src/index.ts]: Done.

**Comment by rgbkrk** [comment]
Addressed initial feedback, this is looking good again. 😄 

screen shot 2019-01-15 at 7 05 04 pm

**Review by captainsafia — APPROVED** [review]
Looks good! Very excited to get this in.

**Comment by rgbkrk** [comment]
🎉

**Pull request merged**